ale/base/data_naif.py

```diff
@@ -322,17 +326,26 @@ def sensor_position(self):
                 # spkezr returns a vector from the observer's location to the aberration-corrected
                 # location of the target. For more information, see:
                 # https://naif.jpl.nasa.gov/pub/naif/toolkit_docs/FORTRAN/spicelib/spkezr.html
-                state, _ = spice.spkezr(self.target_name,
-                                       time,
-                                       self.reference_frame,
-                                       self.light_time_correction,
-                                       self.spacecraft_name,)
-                pos.append(state[:3])
-                vel.append(state[3:])
-            # By default, spice works in km, and the vector returned by spkezr points the opposite
-            # direction to what ALE needs, so it must be multiplied by (-1)
-            self._position = [p * -1000 for p in pos]
-            self._velocity = [v * -1000 for v in vel]
+                if self.swap_observer_target:
+                    state, _ = spice.spkezr(self.target_name,
+                                           time,
+                                           self.reference_frame,
+                                           self.light_time_correction,
+                                           self.spacecraft_name,)
+                    pos.append(-state[:3])
+                    vel.append(-state[3:])
+                else:
+                    state, _ = spice.spkezr(self.spacecraft_name,
+                                           time,
+                                           self.reference_frame,
+                                           self.light_time_correction,
+                                           self.target_name,)
+                    pos.append(state[:3])
+                    vel.append(state[3:])
+
+            # By default, SPICE works in km, so convert to m
+            self._position = [p * 1000 for p in pos]
+            self._velocity = [v * 1000 for v in vel]
```

Comment:
Probably cleaner if you store target and spacecraft in temp vars and make one splezr call using the temp vars. Makes the intent clearer and uses fewer lines.